test(kmc-analyze): add test for Warn_PreviousMapFileCouldNotBeLoaded

cvosoft opened the pull request:
### Summary

This PR adds a unit test for the `Warn_PreviousMapFileCouldNotBeLoaded` message in `AnalyzerMessages`.

### Context

This is related to the "good first issue" about low test coverage in `developer/src/kmc-analyze`.

I couldn't run the tests locally on Linux due to Node.js + ESM + ts-node compatibility issues, but I followed the project's existing patterns closely. Feedback welcome!

mcdurdin reviewed (changes requested):
Thank you for the PR!

Unfortunately this is not quite what we are looking for in testing the messages. What we want to test is that the analyzer itself generates the message when given a relevant input (the message infrastructure itself is tested in the verifyCompilerMessagesObject call). So you could look at some of the other message tests in other kmc modules to see the pattern, for example see kmc-copy: https://github.com/keymanapp/keyman/blob/ba8e5dc1525ebb1d2f2bb6c9372633eafc2a7cf5/developer/src/kmc-copy/test/copier-messages.tests.ts#L33-L38

mcdurdin commented:
> I couldn't run the tests locally on Linux due to Node.js + ESM + ts-node compatibility issues, but I followed the project's existing patterns closely. Feedback welcome!

We don't yet have a docker image for building Developer, which would simplify this for sure. @ermshiperete what do you think? Shouldn't be too hard to do because the build scripts already run on Linux, so probably could start from the Web docker image?

cvosoft commented:
Hi! I tried to follow your suggestion and base the test on `CopierMessages`, but noticed that it also imports `@keymanapp/developer-test-helpers`, which seems to be internal and not part of the public repository or the installable packages.

Many greetings.

mcdurdin commented:
> Hi! I tried to follow your suggestion and base the test on `CopierMessages`, but noticed that it also imports `@keymanapp/developer-test-helpers`, which seems to be internal and not part of the public repository or the installable packages.

Yes, it is an internal package, part of the keyman npm workspace --> developer/src/common/web/test-helpers/package.json. It should resolve automatically if added to the kmc-analyze package.json.

cvosoft commented:
> Thanks, this looks good. One small change requested:
> 
> Rather than poking into the internal method by casting to `any`, can you use the `unitTestEndPoints` pattern as shown here:
> 
> https://github.com/keymanapp/keyman/blob/838aea52ed8c62825030395af6e7ae47f9357cfb/developer/src/kmc-copy/src/KeymanProjectCopier.ts#L716-L720
> 
> That helps with maintenance and code analysis in the future 😁

Thanks for the feedback! I’ve implemented the unitTestEndPoints pattern and cleaned up the test as suggested.

mcdurdin commented on developer/src/kmc-analyze/test/analyzer-messages.tests.ts:
```diff
@@ -26,4 +28,15 @@ describe('AnalyzerMessages', function () {
+
+    const invalidFile = 'nonexistent-file.json';
+
+    analyzer.unitTestEndPoints.loadPreviousMap(invalidFile);
```
```suggestion
    assert.isNull(analyzer.unitTestEndPoints.loadPreviousMap(invalidFile));
```

mcdurdin commented:
Test-bot: skip
